cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -444,44 +456,29 @@ protected SingleOutputSink(BatchSink<IN, KEY_OUT, VAL_OUT> sink,
     }
 
     public void write(IN input) throws Exception {
-      sink.transform(input, emitter);
-      for (KeyValue<KEY_OUT, VAL_OUT> outputRecord : emitter) {
-        context.write(outputRecord.getKey(), outputRecord.getValue());
-      }
-      emitter.reset();
+      KeyValue<KEY_OUT, VAL_OUT> outputRecord = (KeyValue<KEY_OUT, VAL_OUT>) input;
```

Comment:
if we know it's always a KeyValue, we don't need the IN generic.  We also don't need to take the BatchSink as a field.